sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -158,6 +163,18 @@ object GpuDeviceManager extends Logging {
         logWarning(s"Initial RMM allocation(${initialAllocation / 1024 / 1024.0} MB) is " +
           s"larger than free memory(${info.free / 1024 / 1024.0} MB)")
       }
+      val maxAllocation = if (conf.rmmAllocMaxFraction < 1) {
+        (conf.rmmAllocMaxFraction * info.total).toLong
+      } else {
+        // Do not attempt to enforce any artificial pool limit based on queried GPU memory size
+        // if config indicates all GPU memory should be used.
+        0
```

Comment:
We should document this

Reply:
Sorry I should be more clear. In the case of UVM setting it above 1.0 might be logical so we should document in the config that this is happening.

Reply:
I think RMM's `pool_memory_resource` will still query the GPU and use it's reported total memory as a maximum limit internally, although I'm unsure what that API does when UVM is being used.  I'll still add a comment saying the maximum limit is disabled if the specified fraction >= 1.